protoc-gen-gripmock/generator.go

```diff
@@ -10,52 +10,67 @@ import (
 	"strings"
 	"text/template"
 
-	"github.com/gobuffalo/packr/v2"
+	"google.golang.org/protobuf/types/pluginpb"
+
 	"github.com/golang/protobuf/proto"
 	"github.com/golang/protobuf/protoc-gen-go/descriptor"
-	"github.com/golang/protobuf/protoc-gen-go/generator"
-	plugin_go "github.com/golang/protobuf/protoc-gen-go/plugin"
+	"github.com/markbates/pkger"
 	"golang.org/x/tools/imports"
+	"google.golang.org/protobuf/compiler/protogen"
 )
 
 func main() {
-	gen := generator.New()
-	byt, err := ioutil.ReadAll(os.Stdin)
-	if err != nil {
-		log.Fatalf("Failed to read input: %v", err)
+	// Tip of the hat to Tim Coulson
+	// https://medium.com/@tim.r.coulson/writing-a-protoc-plugin-with-google-golang-org-protobuf-cd5aa75f5777
+
+	// Protoc passes pluginpb.CodeGeneratorRequest in via stdin
+	// marshalled with Protobuf
+	input, _ := ioutil.ReadAll(os.Stdin)
+	var request pluginpb.CodeGeneratorRequest
+	if err := proto.Unmarshal(input, &request); err != nil {
+		log.Fatalf("error unmarshalling [%s]: %v", string(input), err)
 	}
 
-	err = proto.Unmarshal(byt, gen.Request)
+	// Initialise our plugin with default options
+	opts := protogen.Options{}
+	plugin, err := opts.New(&request)
 	if err != nil {
-		log.Fatalf("Failed to unmarshal proto: %v", err)
+		log.Fatalf("error initializing plugin: %v", err)
+	}
+
+	protos := make([]*descriptor.FileDescriptorProto, len(plugin.Files))
+	for index, file := range plugin.Files {
+		protos[index] = file.Proto
 	}
 
-	gen.CommandLineParameters(gen.Request.GetParameter())
+	params := make(map[string]string)
```

Comment:
previously we have this one line `gen.CommandLineParameters(gen.Request.GetParameter())` and now we had to manually parse the parameter, care to elaborate?

Reply:
I agree,

The reason for the change is that the original generator lib is deprecated, and is not covered by the protocol buffer compatibility guarantee anymore (more details [here](https://github.com/golang/protobuf/blob/master/protoc-gen-go/generator/generator.go#L5)).

I think there may have been some issue using generator with protogen as well (I can't remember).  I'm happy to have a look at backing this part out if its a deal breaker

Thanks for having a look at my PR :)

https://github.com/golang/protobuf/blob/master/protoc-gen-go/generator/generator.go#L5

Reply:
Thankyou for your detailed explanation. PR accepted :)